libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -692,6 +692,7 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 				solAssert(function.parameterTypes().size() == 1, "");
 				arguments.front()->accept(*this);
 				utils().revertWithStringData(*arguments.front()->annotation().type);
+				utils().popStackElement(*arguments.front()->annotation().type);
```

Comment:
We do not need an explicit pop, since this happens after `revert`, so we only need to tell the compiler how the stack should be adjusted in the following control flow. This should actually be done already in `CompilerUtils::revertWithStringData` with `m_context.adjustStackOffset(_argumentType.sizeOnStack())`.

Because of that, I fear there is another bug lurking below.

Reply:
I was wondering about the same, but since in the `require` case you do the same, I've figured (without looking too much into) that this is how you intended it.

See https://github.com/ethereum/solidity/blob/develop/libsolidity/codegen/ExpressionCompiler.cpp#L941